Add documentation how to avoid fickering windows when running test locally (#215)
napari/docs: docs/developers/testing.md

CI=1 pytest
```

It is also possible to run test using `tox`. This is the same way as it is done in CI.
The main difference is that tox will create a virtual environment for each test environment, so it will take more time
but it will be more similar to the CI environment.

```sh
tox -e py310-linux-pyqt5
```

To get list of all available environments run:

```sh
tox list
```

> ### Running tests without jumping windows

Reviewer: ```suggestion
### Running tests without pop-up windows
```

> Part of tests are opening and closing windows. This can be annoying if you are doing something when wait on the test end. 

Reviewer: ```suggestion
Some tests create visible napari viewers, which pop up on your monitor then quickly disappear.
This can be annoying if you are trying to use your computer while the tests are running.
```

There are two ways to avoid this:

> 1. Use the `QT_QPA_PLATFORM=offscreen` environment variable. This instruct QT to use `offscreen` rendering method.

Reviewer: ```suggestion
1. Use the `QT_QPA_PLATFORM=offscreen` environment variable.
This tells Qt to render windows "offscreen", which is slower but will avoid the pop-ups.
```

   ```shell
   QT_QPA_PLATFORM=offscreen pytest napari
   ```
   or 
   ```shell
   QT_QPA_PLATFORM=offscreen tox -e py310-linux-pyqt5
   ```
   

> 2. Linux only (windows with WSL should work also): use the `xvfb-run` command. This will run the test in a virtual X server. 

Reviewer: ```suggestion
2. If you are using Linux or WSL, you can use the `xvfb-run` command.
   This will run the tests in a virtual X server. 
```

   ```sh
   xvfb-run pytest napari
   ```
   or
   ```sh
   xvfb-run tox -e py310-linux-pyqt5
   ```
   

> where tox environment selector `py310-linux-pyqt5` need to fit your os and python version.

Reviewer: ```suggestion
where the tox environment selector `py310-linux-pyqt5` must match your OS and Python version.
```
